### ISSUE TITLE
System.IO.FileLoadException from the "Get started easily" on the main page

### ISSUE THREAD

**Issue by nikoudel** [fsbolero/Bolero#185]
The very first example from [the home page](https://fsbolero.io/) - "Get started easily" - fails with this exception:
```
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
File name: 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=3.1.8.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
   at MyApp.Server.Startup.ConfigureServices(IServiceCollection services)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.InvokeCore(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass9_0.<Invoke>g__Startup|0(IServiceCollection serviceCollection)
   at Microsoft.AspNetCore.Hosting.StartupLoader.ConfigureServicesDelegateBuilder`1.<>c__DisplayClass15_0.<BuildStartupServicesFilterPipeline>g__RunPipeline|0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.Invoke(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass8_0.<Build>b__0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.StartupLoader.ConfigureServicesDelegateBuilder`1.<>c__DisplayClass14_0.<ConfigureServices>g__ConfigureServicesWithContainerConfiguration|0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.WebHost.Initialize()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at MyApp.Server.Program.main(String[] args) in C:\Users\Valja\git\misc\bolero\MyApp\src\MyApp.Server\Startup.fs:line 53
```

The installed SDK should be fine:
```
dotnet --version
3.1.302
```

**Comment by Tarmil** [comment]
This is odd, I just tried with 3.1.302 and had no issues. I wonder whether different dependencies were resolved somehow. Can you post the output of the following command?

```
dotnet list src/MyApp.Server package --include-transitive
```

**Comment by nikoudel** [comment]
Updated the SDK to 3.1.401 and still getting the same error. Here is the list of transitive deps:
```
C:\git\bolero\MyApp> dotnet list src/MyApp.Server package --include-transitive
Project 'MyApp.Server' has the following package references
   [netcoreapp3.1]:
   Top-level Package                                         Requested   Resolved
   > Bolero.HotReload.Server                                 0.*         0.15.4
   > Bolero.Server                                           0.*         0.15.23
   > FSharp.Core                                             4.7.2       4.7.2
   > Microsoft.AspNetCore.Components.WebAssembly.Server      3.2.*       3.2.1
   > Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation       3.1.*       3.1.8

   Transitive Package                                                                   Resolved
   > Bolero                                                                             0.15.23
   > Bolero.Build                                                                       0.15.23
   > Bolero.HotReload                                                                   0.15.4
   > Elmish                                                                             3.1.0
   > FSharp.SystemTextJson                                                              0.11.13
   > HtmlAgilityPack                                                                    1.8.0
   > Microsoft.AspNetCore.Authorization                                                 3.1.3
   > Microsoft.AspNetCore.Components                                                    3.1.3
   > Microsoft.AspNetCore.Components.Analyzers                                          3.1.3
   > Microsoft.AspNetCore.Components.Forms                                              3.1.3
   > Microsoft.AspNetCore.Components.Web                                                3.1.3
   > Microsoft.AspNetCore.Components.WebAssembly                                        3.2.0
   > Microsoft.AspNetCore.Components.WebAssembly.Build                                  3.2.0
   > Microsoft.AspNetCore.Components.WebAssembly.DevServer                              3.2.1
   > Microsoft.AspNetCore.Components.WebAssembly.HttpHandler                            3.2.0
   > Microsoft.AspNetCore.Components.WebAssembly.Runtime                                3.2.0
   > Microsoft.AspNetCore.Connections.Abstractions                                      3.1.0
   > Microsoft.AspNetCore.Http.Connections.Client                                       3.1.0
   > Microsoft.AspNetCore.Http.Connections.Common                                       3.1.0
   > Microsoft.AspNetCore.Http.Features                                                 3.1.0
   > Microsoft.AspNetCore.Metadata                                                      3.1.3
   > Microsoft.AspNetCore.Mvc.Razor.Extensions                                          3.1.8
   > Microsoft.AspNetCore.Razor.Language                                                3.1.8
   > Microsoft.AspNetCore.SignalR.Client                                                3.1.0
   > Microsoft.AspNetCore.SignalR.Client.Core                                           3.1.0
   > Microsoft.AspNetCore.SignalR.Common                                                3.1.0
   > Microsoft.AspNetCore.SignalR.Protocols.Json                                        3.1.0
   > Microsoft.CodeAnalysis.Analyzers                                                   2.9.3
   > Microsoft.CodeAnalysis.Common                                                      3.3.0
   > Microsoft.CodeAnalysis.CSharp                                                      3.3.0
   > Microsoft.CodeAnalysis.Razor                                                       3.1.8
   > Microsoft.Extensions.Configuration                                                 3.1.3
   > Microsoft.Extensions.Configuration.Abstractions                                    3.1.3
   > Microsoft.Extensions.Configuration.Binder                                          3.1.3
   > Microsoft.Extensions.Configuration.FileExtensions                                  3.1.3
   > Microsoft.Extensions.Configuration.Json                                            3.1.3
   > Microsoft.Extensions.DependencyInjection                                           3.1.3
   > Microsoft.Extensions.DependencyInjection.Abstractions                              3.1.3
   > Microsoft.Extensions.DependencyModel                                               3.1.6
   > Microsoft.Extensions.FileProviders.Abstractions                                    3.1.3
   > Microsoft.Extensions.FileProviders.Physical                                        3.1.3
   > Microsoft.Extensions.FileSystemGlobbing                                            3.1.3
   > Microsoft.Extensions.Http                                                          3.1.0
   > Microsoft.Extensions.Logging                                                       3.1.3
   > Microsoft.Extensions.Logging.Abstractions                                          3.1.3
   > Microsoft.Extensions.Options                                                       3.1.3
   > Microsoft.Extensions.Primitives                                                    3.1.3
   > Microsoft.JSInterop                                                                3.1.3
   > Microsoft.JSInterop.WebAssembly                                                    3.2.0
   > Microsoft.NETCore.Platforms                                                        2.1.2
   > Microsoft.NETCore.Targets                                                          1.1.0
   > Microsoft.Win32.Primitives                                                         4.3.0
   > MyApp.Client                                                                       1.0.0
   > NETStandard.Library                                                                1.6.1
   > runtime.debian.8-x64.runtime.native.System.Security.Cryptography.OpenSsl           4.3.0
   > runtime.fedora.23-x64.runtime.native.System.Security.Cryptography.OpenSsl          4.3.0
   > runtime.fedora.24-x64.runtime.native.System.Security.Cryptography.OpenSsl          4.3.0
   > runtime.native.System                                                              4.3.0
   > runtime.native.System.IO.Compression                                               4.3.0
   > runtime.native.System.Net.Http                                                     4.3.0
   > runtime.native.System.Security.Cryptography.Apple                                  4.3.0
   > runtime.native.System.Security.Cryptography.OpenSsl                                4.3.0
   > runtime.opensuse.13.2-x64.runtime.native.System.Security.Cryptography.OpenSsl      4.3.0
   > runtime.opensuse.42.1-x64.runtime.native.System.Security.Cryptography.OpenSsl      4.3.0
   > runtime.osx.10.10-x64.runtime.native.System.Security.Cryptography.Apple            4.3.0
   > runtime.osx.10.10-x64.runtime.native.System.Security.Cryptography.OpenSsl          4.3.0
   > runtime.rhel.7-x64.runtime.native.System.Security.Cryptography.OpenSsl             4.3.0
   > runtime.ubuntu.14.04-x64.runtime.native.System.Security.Cryptography.OpenSsl       4.3.0
   > runtime.ubuntu.16.04-x64.runtime.native.System.Security.Cryptography.OpenSsl       4.3.0
   > runtime.ubuntu.16.10-x64.runtime.native.System.Security.Cryptography.OpenSsl       4.3.0
   > System.AppContext                                                                  4.3.0
   > System.Buffers                                                                     4.3.0
   > System.Collections                                                                 4.3.0
   > System.Collections.Concurrent                                                      4.3.0
   > System.Collections.Immutable                                                       1.5.0
   > System.Console                                                                     4.3.0
   > System.Diagnostics.Debug                                                           4.3.0
   > System.Diagnostics.DiagnosticSource                                                4.3.0
   > System.Diagnostics.Tools                                                           4.3.0
   > System.Diagnostics.Tracing                                                         4.3.0
   > System.Globalization                                                               4.3.0
   > System.Globalization.Calendars                                                     4.3.0
   > System.Globalization.Extensions                                                    4.3.0
   > System.IO                                                                          4.3.0
   > System.IO.Compression                                                              4.3.0
   > System.IO.Compression.ZipFile                                                      4.3.0
   > System.IO.FileSystem                                                               4.3.0
   > System.IO.FileSystem.Primitives                                                    4.3.0
   > System.IO.Pipelines                                                                4.7.0
   > System.Linq                                                                        4.3.0
   > System.Linq.Expressions                                                            4.3.0
   > System.Memory                                                                      4.5.3
   > System.Net.Http                                                                    4.3.2
   > System.Net.Primitives                                                              4.3.0
   > System.Net.Sockets                                                                 4.3.0
   > System.ObjectModel                                                                 4.3.0
   > System.Reflection                                                                  4.3.0
   > System.Reflection.Emit                                                             4.3.0
   > System.Reflection.Emit.ILGeneration                                                4.3.0
   > System.Reflection.Emit.Lightweight                                                 4.3.0
   > System.Reflection.Extensions                                                       4.3.0
   > System.Reflection.Metadata                                                         1.6.0
   > System.Reflection.Primitives                                                       4.3.0
   > System.Reflection.TypeExtensions                                                   4.3.0
   > System.Resources.ResourceManager                                                   4.3.0
   > System.Runtime                                                                     4.3.0
   > System.Runtime.CompilerServices.Unsafe                                             4.5.2
   > System.Runtime.Extensions                                                          4.3.0
   > System.Runtime.Handles                                                             4.3.0
   > System.Runtime.InteropServices                                                     4.3.0
   > System.Runtime.InteropServices.RuntimeInformation                                  4.3.0
   > System.Runtime.Numerics                                                            4.3.0
   > System.Security.Cryptography.Algorithms                                            4.3.0
   > System.Security.Cryptography.Cng                                                   4.3.0
   > System.Security.Cryptography.Csp                                                   4.3.0
   > System.Security.Cryptography.Encoding                                              4.3.0
   > System.Security.Cryptography.OpenSsl                                               4.3.0
   > System.Security.Cryptography.Primitives                                            4.3.0
   > System.Security.Cryptography.X509Certificates                                      4.3.0
   > System.Text.Encoding                                                               4.3.0
   > System.Text.Encoding.CodePages                                                     4.5.1
   > System.Text.Encoding.Extensions                                                    4.3.0
   > System.Text.Json                                                                   4.7.2
   > System.Text.RegularExpressions                                                     4.3.0
   > System.Threading                                                                   4.3.0
   > System.Threading.Channels                                                          4.7.0
   > System.Threading.Tasks                                                             4.3.0
   > System.Threading.Tasks.Extensions                                                  4.5.3
   > System.Threading.Timer                                                             4.3.0
   > System.ValueTuple                                                                  4.4.0
   > System.Xml.ReaderWriter                                                            4.3.0
   > System.Xml.XDocument                                                               4.3.0
   > System.Xml.XmlDocument                                                             4.3.0
   > System.Xml.XPath                                                                   4.3.0
   > System.Xml.XPath.XmlDocument                                                       4.3.0
   > TaskBuilder.fs                                                                     2.1.0
```

**Comment by Tarmil** [comment]
Okay, I just tried on a machine whose highest installed SDK was 3.1.401, and I got the same error as you. Then I installed 3.1.402, and it worked. So this confirms that the minimum required SDK is 3.1.402.

That's actually the version I recommended in [the blog entry for the latest release](https://forums.websharper.com/blog/88315), but it's good to have the confirmation that it is indeed the absolute minimum required. I'll update the home page right away.
